File: examples/companion-packages/scala-maven-app/src/main/scala/ai/agentxm/examples/pawmatch/MatchPreferences.scala
package ai.agentxm.examples.pawmatch

/** User-supplied preferences for the `match` command. */
final case class MatchPreferences(
    hasKids: Boolean = false,
    quietHome: Boolean = false,
    active: Boolean = false,
    firstTime: Boolean = false,
    multiplePets: Boolean = false,
    smallHome: Boolean = false,
):
  def activeFlagSet: Set[String] =
    val builder = Set.newBuilder[String]
    if hasKids then builder += "has-kids"
    if quietHome then builder += "quiet-home"
    if active then builder += "active"
    if firstTime then builder += "first-time"
    if multiplePets then builder += "multiple-pets"
    if smallHome then builder += "small-home"
    builder.result()

  def isEmpty: Boolean =
    !hasKids && !quietHome && !active && !firstTime && !multiplePets && !smallHome

File: examples/companion-packages/kotlin-maven-app/src/main/kotlin/ai/agentxm/examples/pawmatch/MatchPreferences.kt
package ai.agentxm.examples.pawmatch

data class MatchPreferences(
    val hasKids: Boolean = false,
    val quietHome: Boolean = false,
    val active: Boolean = false,
    val firstTime: Boolean = false,
    val multiplePets: Boolean = false,
    val smallHome: Boolean = false,
) {
    fun activeFlagSet(): Set<String> = buildSet {
        if (hasKids) add("has-kids")
        if (quietHome) add("quiet-home")
        if (active) add("active")
        if (firstTime) add("first-time")
        if (multiplePets) add("multiple-pets")
        if (smallHome) add("small-home")
    }

    fun isEmpty(): Boolean =
        !hasKids && !quietHome && !active && !firstTime && !multiplePets && !smallHome
}
